src/types.ts

```diff
@@ -38,8 +39,22 @@ export type ReaderState = {
   currentTocUrl: string | null;
 };
 
+// Deals with an error where this type is missing from pdfjs
+export type PDFTreeNode = {
+  title: string;
+  bold: boolean;
+  italic: boolean;
+  color: Uint8ClampedArray;
+  dest: string | unknown[] | null;
+  url: string | null;
+  unsafeUrl: string | undefined;
+  newWindow: boolean | undefined;
+  count: number | undefined;
+  items: PDFTreeNode[];
+};
+
 // PDF specific reader state
-export type PdfReaderState = ReaderState;
+export type PdfReaderState = ReaderState & { pdf?: PDFDocumentProxy };
```

Comment:
If I'm not wrong, this should be added inside `/PdfReader/index.tsx` in the `PdfState`. The `PdfReaderState` type here is meant to be the public-facing state that is returned by `usePdfReader`, but I don't think that consumers of `usePdfReader` would want access to that? The `PdfState` is where I keep state that is internal to the component. Confusing, I know. I should do rename these things